libpermian/plugins/anaconda_webui/hypervisor.py

```diff
@@ -0,0 +1,140 @@
+import subprocess
+import time
+import logging
+import os
+import tempfile
+import random
+
+
+START_PORT = '42000'
+LOGGER = logging.getLogger(__name__)
+
+
+class Hypervisor():
+    def __init__(self, host):
+        self.host = host
+
+        if host in ['localhost', '127.0.0.1', '::1']:
+            self.remote = False
+            self.qemu_host = 'qemu:///system'
+        else:
+            self.remote = True
+            self.qemu_host = f'qemu+ssh://{host}/system'
+            self.tempdir = tempfile.TemporaryDirectory(prefix='permian_sshctl.')
+            self.ssh_ctl_file = os.path.join(self.tempdir.name, '%r@%h:%p')
+            self._open_connection()
+
+    def setup(self):
+        # Try to create storage pool boot-scratch to avoid issues later, if it exist its ok for this command to fail.
+        self.virsh_call([
+            'pool-create-as', '--name', 'boot-scratch',
+            '--type', 'dir', '--target' ,'/var/lib/libvirt/boot',
+            '--build'
+        ])
+
+    def forwarding_cleanup(self):
+        self._close_conntection()
+        self.tempdir.cleanup()
+
+    def _open_connection(self):
+        """ Start ssh master connection """
+        subprocess.run(
+            ['ssh', '-S', self.ssh_ctl_file,
+             '-o', 'ControlPersist=yes',
+             '-M', '-N', f'{self.host}'
+            ], check=True)
+
+    def _close_conntection(self):
```

Comment:
Typo in the method name.
Also this method should be called implicitly without need of explicitly calling it. E.g. it could be implicitly called in `__del__` making sure the ssh process is correctly (and gracefully) terminated.
Again, it would be good if this method could be called repeatedly not doing anything if the connection has been closed already.